pkg/build/controller/build/build_controller.go

```diff
@@ -321,7 +322,7 @@ func (bc *BuildController) handleNewBuild(build *buildapi.Build, pod *v1.Pod) (*
 }
 
 // createPodSpec creates a pod spec for the given build
-func (bc *BuildController) createPodSpec(originalBuild *buildapi.Build, ref string) (*v1.Pod, error) {
+func (bc *BuildController) createPodSpec(originalBuild *buildapi.Build, ref string, secret *kapi.LocalObjectReference) (*v1.Pod, error) {
```

Comment:
A comment about why the secret is done here now and not in the instantiate could be some useful context.  Essentially some flavor of https://github.com/openshift/origin/pull/15102#issuecomment-313776269